spec/features/user_signup_functionality_spec.rb

```diff
@@ -0,0 +1,100 @@
+require "rails_helper"
+
+describe 'a new user signin up', type: :feature do
+
+  describe "the user does not already exist" do
+
+    before(:each) do
+      visit "/"
+      click_link("Create Account")
+    end
+
+    it "can create a new account" do
+      fill_in('new_user[email]', with: 'Jeff@gmail.com')
+      fill_in('new_user[user_name]', with: 'Jeffy')
+      fill_in('new_user[password]', with: 'pass')
+      fill_in('new_user[password_confirmation]', with: 'pass')
+      click_button "Create"
+      expect(current_path).to eq(dashboard_path)
+      expect(page).to have_content("Username: Jeffy")
+      expect(page).to have_content("Email: Jeff@gmail.com")
+    end
+
+    it "can not create an account without a username" do
+      fill_in('new_user[email]', with: 'Jeff@gmail.com')
+      fill_in('new_user[password]', with: 'pass')
+      fill_in('new_user[password_confirmation]', with: 'pass')
+      click_button "Create"
+      expect(current_path).to eq('/register')
+      expect(page).to have_content("User name can't be blank")
+    end
+
+    it "can not create an account without an email" do
+      fill_in('new_user[user_name]', with: 'Jeffy')
+      fill_in('new_user[password]', with: 'pass')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.